b2sdk/transfer/emerge/emerger.py

```diff
@@ -130,14 +158,26 @@ def emerge_stream(
 
         :param int min_part_size: lower limit of part size for the transfer planner, in bytes
         :param int max_part_size: upper limit of part size for the transfer planner, in bytes
-
+        :param str,None large_file_sha1: result file SHA1 hash or ``None`` if not known
         """
         planner = self.get_emerge_planner(
             min_part_size=min_part_size,
             recommended_upload_part_size=recommended_upload_part_size,
             max_part_size=max_part_size,
         )
+
+        # ensure that write_intent_iterator is an iterator (and not just iterable, like a list)
+        if not isinstance(write_intent_iterator, Iterator):
```

Comment:
@ppolewicz I remember there was a rule about not using `isinstance` in this project, but I can't recall the reasoning behind it, what was it?

Reply:
@ppolewicz, we're still watiing for your comment here.

Anyway, it seems like `isinstance` is used in 12 other places in this module, not counting what's in the tests.